sycl/test-e2e/Graph/RecordReplay/record_inconsistent_contexts.cpp

```diff
@@ -0,0 +1,27 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out 2>&1 | FileCheck %s
+//
+// CHECK: begin_recording called for a queue whose context differs from the graph context
```

Comment:
Related to the comment about checking error codes, I think checking the exception message is a bit fragile compared to the error code, since it is more likely to change.